### PR TITLE
Fix mongo client factory for reactive projects

### PR DESCRIPTION
**Pull request by juancgalvis**
## Description
Update mongo client factory, remove deprecated instantiation of MongoClientFactory.
Use the correct ReactiveMongoClientFactory for reactive projects.

## Category
- [ ] Feature
- [X] Fix

## Checklist
- [X] The pull request is complete according to the [guide of contributing](https://github.com/bancolombia/scaffold-clean-architecture/wiki/Contributing)
- [ ] Automated tests are written
- [X] The documentation is up-to-date
- [X] the version of the build.gradle, readme.md and gradle.properties was increased
- [X] The pull request has a descriptive title that describes what has changed, and provides enough context for the changelog

### CODE REVIEW

File: src/main/resources/driven-adapter/mongo-repository/config/mongo-config.java.mustache [hunk 1]
```diff
@@ -6,12 +6,14 @@ import co.com.bancolombia.secretsmanager.api.exceptions.SecretException;
 import org.springframework.beans.factory.annotation.Value;
 {{/include-secret}}
 import {{package}}.mongo.config.MongoDBSecret;
-import org.springframework.boot.autoconfigure.mongo.MongoClientFactory;
-import org.springframework.boot.autoconfigure.mongo.MongoProperties;
+import org.springframework.boot.autoconfigure.mongo.*;
```
Comment (santitigaga) [inline src/main/resources/driven-adapter/mongo-repository/config/mongo-config.java.mustache]: you required all class from org.springframework.boot.autoconfigure.mongo?

File: src/main/resources/driven-adapter/mongo-reactive/config/mongo-config.java.mustache [hunk 2]
```diff
@@ -6,12 +6,14 @@ import co.com.bancolombia.secretsmanager.api.GenericManagerAsync;
 import org.springframework.beans.factory.annotation.Value;
 {{/include-secret}}
 import {{package}}.mongo.config.MongoDBSecret;
-import org.springframework.boot.autoconfigure.mongo.MongoClientFactory;
-import org.springframework.boot.autoconfigure.mongo.MongoProperties;
+import org.springframework.boot.autoconfigure.mongo.*;
```
Comment (santitigaga) [inline src/main/resources/driven-adapter/mongo-reactive/config/mongo-config.java.mustache]: you required all class from org.springframework.boot.autoconfigure.mongo?

File: src/main/resources/driven-adapter/mongo-repository/config/mongo-config.java.mustache (hunk 1 above)
Comment (juancgalvis) [reply src/main/resources/driven-adapter/mongo-repository/config/mongo-config.java.mustache]: When a project imports three or more classes from the same package the IDE optimize imports like this.

File: src/main/resources/driven-adapter/mongo-reactive/config/mongo-config.java.mustache (hunk 2 above)
Comment (juancgalvis) [reply src/main/resources/driven-adapter/mongo-reactive/config/mongo-config.java.mustache]: When a project imports three or more classes from the same package the IDE optimize imports like this.

**Review by santitigaga — APPROVED** [review]

**Pull request merged**